### PR TITLE
btn: append quality on release name if not present

### LINKED ISSUE #3852: BTN - Release names with little information, but API returns more
Release names with little information can be problematic. Reading through the code it looks like the BTN provider already tries to do it's best when the ReleaseName is missing; however, some release names do not provide enough information to parse. Since SickRage doesn't / can't use the API information. I propose to check to see if the needed information is available in the ReleaseName. If it is not available, throw the ReleaseName out and use a 'creative' name to return as the title. I can contribute and test these changes since it is a private API and I know Python quite well.

What do you think would be the best conditions to do this in (if at all)?

Check for...
- Resolution only
- Resolution and Source
- Others?

Branch/Commit: develop
OS: FreeBSD 11
Logs:
```
2017-06-10 19:24:41 DEBUG SEARCHQUEUE-BACKLOG-318161 :: [BTN] :: Found episode by absolute number 6 which is S01E06
2017-06-10 19:24:41 DEBUG SEARCHQUEUE-BACKLOG-318161 :: [BTN] :: Ignoring result [Asenshi] Little Witch Academia - 05 [6506D1AB].
2017-06-10 19:24:41 DEBUG SEARCHQUEUE-BACKLOG-318161 :: [BTN] :: Don't want this quality, ignoring found result for Little Witch Academia S01E05 with quality Unknown
2017-06-10 19:24:41 DEBUG SEARCHQUEUE-BACKLOG-318161 :: [BTN] :: Any,Best = [ SDTV, SD DVD, 720p HDTV, 1080p HDTV, 720p WEB-DL, 1080p WEB-DL, 720p BluRay, 1080p BluRay, 4K UHD TV, 4K UHD WEB-DL, 4K UHD BluRay, 8K UHD TV, 8K UHD WEB-DL, 8K UHD BluRay ] [ None ] Found = [ Unknown ]
2017-06-10 19:24:41 DEBUG SEARCHQUEUE-BACKLOG-318161 :: [BTN] :: Parsed [Asenshi] Little Witch Academia - 05 [6506D1AB] into Little Witch Academia - S01E05 [ABS: [5]] [ANIME VER: 1] [GROUP: Asenshi] [ABD: False] [ANIME: True] [whichReg: [u'anime_ultimate']]
```

### PR DESCRIPTION
**Pull request by nopjmp**
This should prevent problems when release names do not include the quality for some reason on BTN.

Additionally clean up the results to not return results that had invalid
json returned while we are here.

Fixes #3852

Proposed changes in this pull request:
- Append quality if not present
- make 'creative' title None by default
- Add check when title or url is None. This was done prior in another place already.

- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read [contribution guide](https://github.com/SickRage/SickRage/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: sickbeard/providers/btn.py [hunk 1]
```diff
@@ -172,12 +172,22 @@ def _get_title_and_url(self, parsed_json):
...
+            if 'Codec' in parsed_json and parsed_json['Codec'].lower() not in title.lower():
+                append += parsed_json['Codec']
+
+            if len(append) > 0
+                title += ' [' + append + ']'
 
         else:
             # If we don't have a release name we need to get creative
-            title = ''
+            title = None
```
Comment (sharkykh) [inline sickbeard/providers/btn.py]: Why did you change this? Empty string **is** falsey (`bool(None) == bool('') == False`)

File: sickbeard/providers/btn.py [hunk 2]
```diff
@@ -172,12 +172,22 @@ def _get_title_and_url(self, parsed_json):
...
             if 'Series' in parsed_json:
-                title += parsed_json['Series']
+                title = parsed_json['Series']
```
Comment (sharkykh) [inline sickbeard/providers/btn.py]: Related to the comment above,
that could possibly cause exceptions
if for some odd reason `parsed_json['Series']` isn't available,

**Review by sharkykh — COMMENTED** [review]
Thank you for the contribution.
Please see my comments.

File: sickbeard/providers/btn.py [hunk 3]
```diff
@@ -172,12 +172,22 @@ def _get_title_and_url(self, parsed_json):
...
             title = parsed_json['ReleaseName']
+            append = ''
+            if 'Resolution' in parsed_json and parsed_json['Resolution'].lower() not in title.lower():
+                append += parsed_json['Resolution']
+            if 'Source' in parsed_json and parsed_json['Source'].lower() not in title.lower():
+                append += parsed_json['Source']
+            if 'Codec' in parsed_json and parsed_json['Codec'].lower() not in title.lower():
+                append += parsed_json['Codec']
+
+            if len(append) > 0
```
Comment (sharkykh) [inline sickbeard/providers/btn.py]: You're missing a colon.
BTW, the check can be truthy:
```python
if append:
```
No need to calculate the string length.

**Review by sharkykh — CHANGES_REQUESTED** [review]

File: sickbeard/providers/btn.py (hunk 1 above)
Comment (nopjmp) [reply sickbeard/providers/btn.py]: I forgot python semantics did this. I'm know way too many programming languages. I'll get this reverted!

**Review by sharkykh — APPROVED** [review]
